src/react-apps/applications/ux-editor/src/containers/WorkflowStep.tsx

```diff
@@ -1,56 +1,95 @@
 import * as React from 'react';
 import { connect } from 'react-redux';
+import { Redirect } from 'react-router';
+import FormFillerActionDispatchers from '../actions/formFillerActions/formFillerActionDispatcher';
 
 export interface IWorkflowStepProvidedProps {
   header: string;
+  step: WorkflowSteps;
+  onStepChange: any;
+}
+
+/*
+  Reflects enum at server side
+*/
+export enum WorkflowSteps {
+  Unknown = 0,
+  FormFilling = 1,
+  Submit = 2,
+  Archived = 3,
 }
 
 export interface IWorkflowStepProps extends IWorkflowStepProvidedProps {
   language: any;
   errorList: string[];
 }
 
-export interface IWorkflowStepState {}
+export interface IWorkflowStepState {
+  redirect: boolean;
+  isRuntime: boolean;
+  workflowStep: WorkflowSteps;
+}
 
-class WorkflowStepComponent extends React.Component<IWorkflowStepProps, any> {
+class WorkflowStepComponent extends React.Component<IWorkflowStepProps, IWorkflowStepState> {
+  constructor(props: IWorkflowStepProps, state: IWorkflowStepState) {
+    super(props, state);
+    const isRuntime = window.location.pathname.split('/')[1].toLowerCase() === 'runtime';
+    this.state = {
+      redirect: false,
+      isRuntime,
+      workflowStep: props.step,
+    };
+  }
+
+  public componentWillReceiveProps(nextProps: IWorkflowStepProps) {
+    if (nextProps.step !== this.state.workflowStep) {
+      this.setState({
+        workflowStep: nextProps.step,
+      });
+    }
+  }
 
   public renderTop = () => {
     return (
       <div className='row'>
-          <div className='col-xl-12'>
-            <div className='a-modal-top'>
-              <img
-                src='/designer/img/a-logo-blue.svg'
-                alt='Altinn logo'
-                className='a-logo a-modal-top-logo '
-              />
-              <div className='a-modal-top-user'>
-                <div className='a-personSwitcher ' title={this.props.language.ux_editor.formfiller_placeholder_user}>
-                  <span className='a-personSwitcher-name'>
-                    <span className='d-block' style={{color: '#022F51'}}>
-                      {this.props.language.ux_editor.formfiller_placeholder_user}
-                    </span>
-                    <span className='d-block'/>
+        <div className='col-xl-12'>
+          <div className='a-modal-top'>
+            <img
+              src='/designer/img/a-logo-blue.svg'
+              alt='Altinn logo'
+              className='a-logo a-modal-top-logo '
+            />
+            <div className='a-modal-top-user'>
+              <div className='a-personSwitcher ' title={this.props.language.ux_editor.formfiller_placeholder_user}>
+                <span className='a-personSwitcher-name'>
+                  <span className='d-block' style={{ color: '#022F51' }}>
+                    {this.props.language.ux_editor.formfiller_placeholder_user}
```

Comment:
I guess this key can also be moved to the form_filler section 😄

Reply:
Fixed.